Flaky ruler test TestAPI_CreateRuleGroup

From https://github.com/grafana/mimir/actions/runs/5731385617/job/15532202719?pr=5585

```
 --- FAIL: TestAPI_CreateRuleGroup (1.96s)
    --- FAIL: TestAPI_CreateRuleGroup/with_valid_rules_with_enabled_federation (1.20s)
        logging.go:33: instance localhost level info msg cleaning up mapped rules directory path /tmp/TestAPI_CreateRuleGroup2796865914/002
        logging.go:33: instance localhost level info msg ring doesn't exist in KV store yet
        logging.go:33: instance localhost level info msg instance not found in the ring instance localhost ring ruler
        logging.go:33: instance localhost level info msg waiting until ruler is JOINING in the ring
        logging.go:33: instance localhost level info msg ruler is JOINING in the ring
        logging.go:33: instance localhost level info msg syncing rules reason initial
        logging.go:33: instance localhost level info msg waiting until ruler is ACTIVE in the ring
        logging.go:33: instance localhost level info msg ruler is ACTIVE in the ring
        logging.go:33: instance localhost level info msg ruler is only now starting to evaluate rules
        logging.go:33: instance localhost level info msg ruler up and running
        logging.go:33: instance localhost level info msg syncing rules reason api-change
        logging.go:33: instance localhost level info msg updating rule file file /tmp/TestAPI_CreateRuleGroup2796865914/002/user1/namespace
        logging.go:33: instance localhost level info component ruler insight true user user1 msg Starting rule manager...
        api_test.go:1064: expected <nil>, got 
            
            Diff:
            --- metric output does not match expectation; want
            +++ got:
            @@ -2,3 +2,3 @@
             # TYPE cortex_ruler_sync_rules_total counter
            -cortex_ruler_sync_rules_total{reason="api-change"} 0
            +cortex_ruler_sync_rules_total{reason="api-change"} 1
             cortex_ruler_sync_rules_total{reason="initial"} 1
        logging.go:33: instance localhost level info msg ring lifecycler is shutting down ring ruler
        logging.go:33: instance localhost level info user user1 msg Stopping notification manager...
        logging.go:33: instance localhost level info msg stopping user managers
        logging.go:33: instance localhost level info component ruler insight true user user1 msg Stopping rule manager...
        logging.go:33: instance localhost level info msg unregistering instance from ring ring ruler
        logging.go:33: instance localhost level info msg instance removed from the ring ring ruler
        logging.go:33: instance localhost level info component ruler insight true user user1 msg Rule manager stopped
        logging.go:33: instance localhost level info msg all user managers stopped
        logging.go:33: instance localhost level info msg cleaning up mapped rules directory path /tmp/TestAPI_CreateRuleGroup2796865914/002
level=info msg="updating rule file" file=/rules/user1/file%20%2Fone
level=info msg="updating rule file" file=/rules/user1/file%20%2Fone
level=info msg="updating rule file" file=/rules/user1/file%20%2Fone
level=info msg="updating rule file" file=/rules/user1/file%20%2Ftwo
level=info msg="updating rule file" file=/rules/user1/file%20%2Ftwo
level=info msg="updating rule file" file=/rules/user1/+A_%2FReallyStrange%3C%3ENAME:SPACE%2F%3F
```

--- a/pkg/ruler/api_test.go
+++ b/pkg/ruler/api_test.go
@@ -1052,6 +1052,11 @@ rules:
 			router := mux.NewRouter()
 			router.Path("/prometheus/config/v1/rules/{namespace}").Methods("POST").HandlerFunc(a.CreateRuleGroup)
 			router.Path("/prometheus/config/v1/rules/{namespace}/{groupName}").Methods("GET").HandlerFunc(a.GetRuleGroup)
+
+			// Pre-condition check: the ruler should have run the initial rules sync but not done a sync
+			// based on API mutations.
+			verifySyncRulesMetric(t, reg, 1, 0)
+
 			// POST
 			req := requestFor(t, http.MethodPost, "https://localhost:8080/prometheus/config/v1/rules/namespace", strings.NewReader(tt.input), "user1")
 			w := httptest.NewRecorder()
@@ -1060,9 +1065,6 @@ rules:
 			require.Equal(t, tt.status, w.Code)
 
 			if tt.err == nil {
-				// Pre-condition check: the ruler should have run the initial rules sync.
-				verifySyncRulesMetric(t, reg, 1, 0)
-
 				// GET
 				req = requestFor(t, http.MethodGet, "https://localhost:8080/prometheus/config/v1/rules/namespace/test", nil, "user1")
 				w = httptest.NewRecorder()

